types/bootstrap/js/dist/dropdown.d.ts

```diff
@@ -78,7 +78,7 @@ declare namespace Dropdown {
 
     type OffsetFunction = () => Offset;
 
-    type PopperConfigFunction = () => Partial<Popper.Options>;
+    type PopperConfigFunction = (defaultBsPopperConfig: Partial<Popper.Options>) => Partial<Popper.Options>;
 
     interface Options {
```

Comment:
Here please use 
```suggestion
    interface Options extends Pick<Tooltip.Options, "popperConfig">
```

and remove `popperConfig` and `PopperConfigFunction` from this file